core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -742,9 +752,38 @@ class SnappyParser(session: SnappySession)
         UnresolvedRelation(r), input.sliceString(0, input.length)))
   }
 
+  // Only when wholeStageEnabled try for tokenization. It should be
+  // true
+  private var tokenize = session.sessionState.conf.wholeStageEnabled
+
+  private var isselect = false
+
+  private def setNoTokenize: Boolean = {
+    tokenize = false
+    true
+  }
+
+  private def setTokenize: Boolean = {
+    isselect match {
+      case true => tokenize = session.sessionState.conf.wholeStageEnabled
+      case _ => tokenize = false
+    }
+    true
+  }
+
+  private def setSelect: Boolean = {
+    isselect = true
+    true
+  }
+
+  private def setNoSelect: Boolean = {
+    isselect = false
+    true
+  }
+
   override protected def start: Rule1[LogicalPlan] = rule {
-    query.named("select") | insert | put | dmlOperation | ctes |
-        ddl | set | cache | uncache | desc
+    (test(setSelect) ~ query.named("select")) | (test(setNoSelect) ~ (insert | put | dmlOperation | ctes |
```

Comment:
We just need to align the options in the start method of of the parser. We can do it easily as far as parsing is considered. Tokenizing put and other dmls will, as you say, might come later if at all.